src/emu/emuopts.h

```diff
@@ -195,56 +195,77 @@
 //  TYPE DEFINITIONS
 //**************************************************************************
 
+struct game_driver;
+class device_slot_interface;
+class emu_options;
+
 class slot_option
 {
 public:
-	slot_option(const char *default_value = nullptr);
-	slot_option(const slot_option &that) = default;
+	slot_option(emu_options &host, const char *default_value);
+	slot_option(const slot_option &that) = delete;
 	slot_option(slot_option &&that) = default;
 
-	const slot_option &operator=(const slot_option &that)
-	{
-		m_specified = that.m_specified;
-		m_specified_value = that.m_specified_value;
-		m_specified_bios = that.m_specified_bios;
-		m_default_card_software = that.m_default_card_software;
-		m_default_value = that.m_default_value;
-		return *this;
-	}
-
-	const slot_option &operator=(slot_option &&that)
-	{
-		m_specified = that.m_specified;
-		m_specified_value = std::move(that.m_specified_value);
-		m_specified_bios = std::move(that.m_specified_bios);
-		m_default_card_software = std::move(that.m_default_card_software);
-		m_default_value = std::move(that.m_default_value);
-		return *this;
-	}
-
 	// accessors
 	const std::string &value() const;
 	std::string specified_value() const;
 	const std::string &bios() const { return m_specified_bios; }
 	const std::string &default_card_software() const { return m_default_card_software; }
 	bool specified() const { return m_specified; }
+	core_options::entry *option_entry() const { return m_entry; }
 
 	// seters
 	void specify(std::string &&text);
 	void set_bios(std::string &&text);
-	void set_default_card_software(std::string &&s) { m_default_card_software = std::move(s); }
+	void set_default_card_software(std::string &&s);
+
+	// instantiates an option entry (don't call outside of emuopts.cpp)
+	core_options::entry::ptr setup_option_entry(const char *name);
 
 private:
-	bool            m_specified;
-	std::string     m_specified_value;
-	std::string     m_specified_bios;
-	std::string     m_default_card_software;
-	std::string     m_default_value;
+	void possibly_changed(const std::string &old_value);
+
+	emu_options &			m_host;
+	bool					m_specified;
+	std::string				m_specified_value;
+	std::string				m_specified_bios;
+	std::string				m_default_card_software;
+	std::string				m_default_value;
+	core_options::entry *	m_entry;
```

Comment:
I don't think this is particularly safe.  The option allocates a `core_options::entry` and hands it off somewhere as a `unique_ptr`, keeping a naked pointer to it.  For proper safety it should hand off a `shared_ptr` and keep a `weak_ptr`.  The current code is an object lifecycle issue waiting to happen.